### PR TITLE
MAINT: Replace in1d with isin

### PR DESCRIPTION
**Pull request by larsoner**
No idea why we haven't hit this error that mne-qt-browser is hitting:

https://github.com/mne-tools/mne-qt-browser/actions/runs/6155800978/job/16721232893?pr=194

```
../mne-python/mne/viz/_figure.py:376: in _process_data
1276
    np.in1d(this_names, self.mne.whitened_ch_names),
1277
/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/numpy/lib/_arraysetops_impl.py:611: in in1d
1278
    warnings.warn(
1279
E   DeprecationWarning: `in1d` is deprecated. Use `np.isin` instead.
```

But this is a naive replace-in-files, let's see if it works.

### CODE REVIEW

**Comment by larsoner** [comment]
... I'll look into why `numpy`'s pre isn't being used here. Probably some other dependency we install requires `numpy<2` and we should remove it from the `pip-pre` job until it gets updated.

**Comment by larsoner** [comment]
Confirmed that we now have on GHA:
```
├☑ numpy                2.0.0.dev0+git20230908.34d6693 (OpenBLAS 0.3.23.dev with 1 thread)
```
and Azure
```
├☑ numpy                2.0.0.dev0+git20230908.34d6693 (OpenBLAS 0.3.23.dev with 2 threads)
```
So I'll mark for merge-when green!

File: mne/beamformer/tests/test_dics.py [hunk 1]
```diff
@@ -108,7 +108,7 @@ def _simulate_data(fwd, idx):  # Somewhere on the frontal lobe by default
     csd = csd_morlet(epochs, frequencies=[10, 20], n_cycles=[5, 10], decim=5)
 
     labels = mne.read_labels_from_annot("sample", hemi="lh", subjects_dir=subjects_dir)
-    label = [label for label in labels if np.in1d(source_vertno, label.vertices)[0]]
+    label = [label for label in labels if np.isin(source_vertno, label.vertices)]
     assert len(label) == 1
```
Comment (wmvanvliet) [inline mne/beamformer/tests/test_dics.py:112]: removal of `[0]` intentional?

Comment (larsoner) [reply mne/beamformer/tests/test_dics.py:112]: Yes and necessary. source_vertno is a 0d scalar which apparently behaves differently (better I think) for isin compared to in1d

**Pull request merged**